apps/src/code-studio/components/pairing/StudentSelector.jsx

```diff
@@ -47,28 +69,44 @@ export default class StudentSelector extends React.Component {
     } else if (this.props.students.length === 0) {
       return <span>{i18n.noStudentsInSection()}</span>;
     }
-
-    const studentDivs = this.props.students.map(student => {
-      let className = 'student selectable';
+    const exceededMaximum = this.state.selectedStudentIds.length >= 4;
+    const studentBtns = this.props.students.map(student => {
+      let className = 'selectable';
       if (this.state.selectedStudentIds.indexOf(student.id) !== -1) {
```

Comment:
It looks like swapping the logic around here might simplify things a bit:
```
let className = 'selected';
let disabled = false;
if (this.state.selectedStudentIds.indexOf(student.id) >= 0) {
  className = 'selectable';
  disabled = excededMaximum;
}
```

Reply:
This logic doesn't quite line up with what I originally had. We want to change the className to 'selected' if the condition for `if (this.state.selectedStudentIds.indexOf(student.id) >= 0)` evaluates to true, not the other way around. The original condition of `!==` is essentially the same as `>=0`, so flipping the classNames actually disrupts the logic.

I've changed the code to look like this: 
```
let className = 'selected'; 
let disabled = false; 
if (this.state.selectedStudentIds.indexOf(student.id) === -1) {
    className = 'selectable'; 
    disabled = exceededMaximum;
}
```

Let me know if you think this is an improvement. Apologies if I've missed the point of your original comment – I want to make sure it's as clear as possible.

Reply:
This looks great! Sorry for the confusion. You're completely right. I did mis-read i :)